mcp-servers/snow/src/snow/servicenow/client.py

```diff
@@ -0,0 +1,166 @@
+    ServerConfig,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class ServiceNowClient:
+    """
+    ServiceNow API client for making requests to ServiceNow instance.
+    """
+
+    def __init__(self):
+        """
+        Initialize the ServiceNow client with configuration from environment variables.
+        """
+        self.config = self._load_config_from_env()
+        self.auth_manager = AuthManager(self.config.auth, self.config.instance_url)
+
+    def _load_config_from_env(self) -> ServerConfig:
+        """
+        Load configuration from environment variables.
+
+        Returns:
+            ServerConfig: Configuration loaded from environment variables.
+
+        Raises:
+            ValueError: If required environment variables are missing.
+        """
+        instance_url = os.getenv("SERVICENOW_INSTANCE_URL")
+        if not instance_url:
+            raise ValueError("SERVICENOW_INSTANCE_URL environment variable is required")
+
+        auth_type = os.getenv("SERVICENOW_AUTH_TYPE", "basic").lower()
+
+        if auth_type == "basic":
+            username = os.getenv("SERVICENOW_USERNAME")
+            password = os.getenv("SERVICENOW_PASSWORD")
+            if not username or not password:
+                raise ValueError(
+                    "SERVICENOW_USERNAME and SERVICENOW_PASSWORD are required for basic auth"
+                )
+
+            auth_config = AuthConfig(
+                type=AuthType.BASIC,
+                basic=BasicAuthConfig(username=username, password=password),
+            )
+
+        elif auth_type == "oauth":
+            client_id = os.getenv("SERVICENOW_CLIENT_ID")
+            client_secret = os.getenv("SERVICENOW_CLIENT_SECRET")
+            username = os.getenv("SERVICENOW_USERNAME")
+            password = os.getenv("SERVICENOW_PASSWORD")
+
+            if not client_id or not client_secret:
+                raise ValueError(
+                    "SERVICENOW_CLIENT_ID and SERVICENOW_CLIENT_SECRET are required for OAuth"
+                )
+
+            auth_config = AuthConfig(
+                type=AuthType.OAUTH,
+                oauth=OAuthConfig(
+                    client_id=client_id,
+                    client_secret=client_secret,
+                    username=username or "",
+                    password=password or "",
+                    token_url=os.getenv("SERVICENOW_TOKEN_URL"),
+                ),
+            )
+
+        elif auth_type == "api_key":
+            api_key = os.getenv("SERVICENOW_API_KEY")
+            if not api_key:
+                raise ValueError("SERVICENOW_API_KEY is required for API key auth")
+
+            auth_config = AuthConfig(
+                type=AuthType.API_KEY,
+                api_key=ApiKeyConfig(
+                    api_key=api_key,
+                    header_name=os.getenv(
+                        "SERVICENOW_API_KEY_HEADER", "X-ServiceNow-API-Key"
+                    ),
+                ),
+            )
```

Comment:
Not critical, just an idea... we could introduce some sub classes (one for each if condition) for instance --- this could help in the long term maintenance.

Reply:
One questions on that front is how much of this code is a direct copy of the original external repo. If its almost a 100% copy we might want to keep them consistent.